### ISSUE TITLE
Allow Operation-Location with relative URI within OperationResourcePollingStrategy

### ISSUE THREAD

**Issue by martinzmw** [Azure/azure-sdk-for-java#28574]
Hi,

I am using the [Java SDK for the new Azure Device Update for IoT-Hub (ADU)](https://docs.microsoft.com/en-us/java/api/com.azure.iot.deviceupdate?view=azure-java-preview&preserve-view=true). With this API, you can, e.g., import an update into an ADU Instance which will be a so-called long running operation. The underlying HTTP Request is immediately confirmed with a HTTP Response containing the field Operation-Location within the header. The actual import of the update will take some time and carried out later on.

The respective method [`DeviceUpdateClient.beginImportUpdate()`](https://docs.microsoft.com/en-us/java/api/com.azure.iot.deviceupdate.deviceupdateclient.beginimportupdate?view=azure-java-preview) returns a SyncPoller, so that, I can poll the status of the long running operation. That's the theory.

Unfortunately, it doesn't work because the ADU Service returns a **relative URI** instead of an absolute URI in the HTTP header field Operation-Location. And this results in a MalformedURLException at this point:
https://github.com/Azure/azure-sdk-for-java/blob/a4ccbcaf6987e4a16f38a17a0e38433743ba09e1/sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java#L86

Here is a screenshot of a response when carrying out the same action using the REST API: 
![malformed_Operation-Location](https://user-images.githubusercontent.com/98043091/165777751-a5bf238b-83a2-47f5-93fa-eadeb14c3642.png)

Of course, the surrounding method [`OperationResourcePollingStrategy.canPoll()`](https://github.com/Azure/azure-sdk-for-java/blob/8334e5951b3c5ccc35aa4c27005710de7d3b5b47/sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java#L82) catches the MalformedURLException  and returns false. But, this does not  help. The respective instance of `OperationResourcePollingStrategy` is part of an instance of [`DefaultPollingStrategy`](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/util/polling/DefaultPollingStrategy.java). This default strategy tries as next:
* [`LocationPollingStrategy.canPoll()`](https://github.com/Azure/azure-sdk-for-java/blob/8334e5951b3c5ccc35aa4c27005710de7d3b5b47/sdk/core/azure-core/src/main/java/com/azure/core/util/polling/LocationPollingStrategy.java#L82) --> this returns false as well because the HTTP header field Location is also a **relative URI** instead of an absolute URI.
* [`StatusCheckPollingStrategy.canPoll()`](https://github.com/Azure/azure-sdk-for-java/blob/8334e5951b3c5ccc35aa4c27005710de7d3b5b47/sdk/core/azure-core/src/main/java/com/azure/core/util/polling/StatusCheckPollingStrategy.java#L47) --> this returns true, but after calling the `SyncPoller.poll()` method in my application afterwards, the [`StatusCheckPollingStrategy.poll()`](https://github.com/Azure/azure-sdk-for-java/blob/8334e5951b3c5ccc35aa4c27005710de7d3b5b47/sdk/core/azure-core/src/main/java/com/azure/core/util/polling/StatusCheckPollingStrategy.java#L69) method throws an exception as expected.

Here is a short excerpt of my application. Sorry, I can't provide a minimal example because this will require some Azure resources.
```
      BinaryData requestBody = BinaryData.fromString("<the actual import manifest");
      RequestOptions requestOptions = new RequestOptions().addQueryParam("action", "import");
      SyncPoller<BinaryData,BinaryData> syncPoller = client.beginImportUpdate(requestBody, requestOptions);
      PollResponse<BinaryData> pollResponse;
      pollResponse = syncPoller.poll(); // This fails, because the Operation-Location inside the header of the initial response is malformed!
```

I have already contacted the Support using the Azure Portal which delegated my request to the product team. But, they said that the **relative URI** in Operation-Location is just fine.

Could you please enhance the implementation of [`OperationResourcePollingStrategy.canPoll()`](https://github.com/Azure/azure-sdk-for-java/blob/8334e5951b3c5ccc35aa4c27005710de7d3b5b47/sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java#L82) so that it also accepts relative URIs, by prepending the hostname from the respective HTTP request.

Thank you very much,
Martin

**Comment by SaurabhSharma-MSFT** [comment]
@martinzmw We are looking into it and get back to you for any additional information.

**Comment by SatishBoddu-MSFT** [comment]
Hello @martinzmw We are routing this to the Service Teams' attention.

**Comment by alzimmermsft** [comment]
This should be resolved by #29676 and was made available to all Azure SDKs for Java using `azure-core` `1.32.0` or later. For `azure-iot-deviceupdate` this will be version `1.0.0` or newer.
